### PR TITLE
dd: implement byte suffixes, iseek + oseek flags

### PR DESCRIPTION
**Pull request by chordtoll**
These changes allow our implementation to pass the dd/bytes.sh tests:
- Numbers with B suffix (i.e. 10B) force byte rather than record count
- Add iseek and oseek options (additive with skip and seek options)
- Implement tests for above changes
- Implement dd/bytes.sh equivalent tests in test_dd

### CODE REVIEW

**Comment by chordtoll** [comment]
Updated to fix spelling/format issues

File: src/uucore/src/lib/parser/parse_size.rs [hunk 1]
```diff
@@ -55,7 +55,7 @@ pub fn parse_size(size: &str) -> Result<u64, ParseSizeError> {
     // empty string, in which case, the factor is 1.
     let unit = &size[numeric_string.len()..];
     let (base, exponent): (u128, u32) = match unit {
-        "" => (1, 0),
+        "B" | "" => (1, 0),
```
Comment (jfinkels) [inline src/uucore/src/lib/parser/parse_size.rs:58]: Looks like this change is causing a failure in `truncate`, which seems to disallow this:

```
failures:

---- test_truncate::test_invalid_numbers stdout ----
current_directory_resolved: 
run: /home/runner/work/coreutils/coreutils/target/debug/coreutils truncate -s 0X file
current_directory_resolved: 
run: /home/runner/work/coreutils/coreutils/target/debug/coreutils truncate -s 0XB file
current_directory_resolved: 
run: /home/runner/work/coreutils/coreutils/target/debug/coreutils truncate -s 0B file
thread 'test_truncate::test_invalid_numbers' panicked at 'Command was expected to fail.
stdout = 
 stderr = ', tests/common/util.rs:185:9
```

**Review by jfinkels — COMMENTED** [review]
`iseek` and `oseek` are not documented: https://www.gnu.org/software/coreutils/manual/html_node/dd-invocation.html and they don't seem to be supported by the version of coreutils I have on my system, v8.30. Is this a recent feature in GNU `dd`?

As for the `B` suffix, maybe we could split this into two separate pull requests and deal with the two new features separately: one for supporting the `B` suffix and one for adding `iseek` and `oseek`?

**Comment by chordtoll** [comment]
Makes sense to split this PR into one for the flags and one for the suffixes. Closing this one to create those two.

**Pull request closed without merging**